src/pages/react-redux-pitfalls-and-best-pratices/index.md

```diff
@@ -0,0 +1,482 @@
+---
+title: 'React/Redux: pitfalls and best practices'
+description:
+  We use React and Redux for almost 4 years at Bedrock on our video platform named 6play.
+  Good practices and mistakes to be avoided have been gathered in this article.
+date: 2020-04-04
+hero: ./redux.jpg
+language: en
+---
+
+After 2 years using React with Redux for the video platform [6play](https://6play.fr), I was able to identify good practices and pitfalls to avoid at all costs.
+The [Bedrock](https://www.bedrockstreaming.com/) team (to which I currently belong) kept the technical stack of the project up to date to take advantage of the new features of `react`, `react-redux` and `redux`.
+
+So here are my tips for maintaining and using React and Redux in your application without going mad.
+
+_This article is not an introduction to React or Redux. I recommend [this documentation](https://redux.js.org/basics/usage-with-react) if you want to see how to implement it in your applications._
+
+You could also take a look at [Redux offical style guide] in which you could find some of those tips and others.
+
+## Avoid having only one reducer
+
+The [reducer] is the function that is in charge of building a new state at each `action`.
+One might be tempted to manipulate only one reducer.
+In the case of a small application, this is not a problem.
+For applications expressing a complex and evolving business, it is better to opt in for the [combineReducers] solution.
+
+This feature of `redux` allows to manipulate not one but several [reducer]s which act respectively on the state.
+
+> When and how to split its application?
+
+What I recommend is a functional splitting of the application.
+In my approach, I would tend to represent the business of the application more than the technical stuff implied.
+Some very good articles explain it notably through the use of [DDD principles](https://en.wikipedia.org/wiki/Domain-driven_design).
+
+In Bedrock, we use a folder named _modules_ which groups together the different folders associated with the feature of your application.
+
+```txt
+app/
+  modules/
+    user/
+      __tests__/
+        user.reducer.spec.js
+      components/
+      user.reducer.js
+    product/
+      __tests__/
+        product.reducer.spec.js
+      components/
+      product.reducer.js
+    account/
+      __tests__/
+      account.reducer.spec.js
+      components/
+      account.reducer.js
+  store.js
+  index.js
+```
+
+So in `store.js` all you need to do is combine your different reducers.
+
+```js
+import { createStore, combineReducers } from 'redux'
+import { user } from './modules/user/user.reducer.js'
+import { product } from './modules/user/product.reducer.js'
+import { account } from './modules/user/account.reducer.js'
+
+export const store = createStore(combineReducers({ user, product, account }))
+```
+
+By following this principle, you can:
```

Comment:
`you can` => `you will`
i would translate that in : `Tu peux / Vous pouvez` => `Tu vas / Vous allez`.
What do you think?